fetch/fetch.go

```diff
@@ -554,6 +563,40 @@ func Fetch(
 	return nil
 }
 
+func checkForColumnMismatches(logger zerolog.Logger, tables []tableverify.Result) error {
+	hasFailableMismatch := false
+	hasNonFailableMismatch := false
+
+	for _, table := range tables {
+		for _, col := range table.MismatchingTableDefinitions {
+			isFailableMismatch := inconsistency.IsFailableMismatch(col.Tag)
+			if isFailableMismatch {
+				hasFailableMismatch = true
+			} else {
+				hasNonFailableMismatch = true
+			}
+
+			// Don't fail right away so that we can log out all of the
+			// table column mismatches for the user to fix on their end.
+			logger.Error().
+				Str("mismatch_tag", col.Tag).
+				Bool("failable_mismatch", isFailableMismatch).
+				Str("reason", col.Info).
+				Msgf("not migrating column from %s.%s as it mismatches on the target side", col.Name.Schema, col.Name.Table)
+		}
+	}
+
+	if hasFailableMismatch {
+		return errors.Newf(`exiting early due to mismatching column definitions that must be manually resolved. For details, see logs with "mismatch_tag".`)
+	}
+
+	if hasNonFailableMismatch {
+		logger.Warn().Msg(`found only column mismatches that do not block migration. For details, see logs with "mismatch_tag". Fetch continuing.`)
```

Comment:
I feel like this is simplest, unless it is important to say that "only" non-failing mismatches were found.

```suggestion
		logger.Warn().Msg(`found column mismatches. For details, see logs with "mismatch_tag". Fetch continuing.`)
```